### PR TITLE
Document lessons learned during the release of 0.3.2 [skip ci]

### PR DESCRIPTION
**Pull request by porunov**
Signed-off-by: Oleksandr Porunov <alexandr.porunov@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -16,6 +15,38 @@ The release artifacts will be deployed into Sonatype OSS with the Maven deploy p
...
+The command to publish your keys to a key server is the next:
+```Shell
+gpg --keyserver <key server> --send-keys <key id>
+```
+An example: 
+```Shell
+gpg --keyserver pgp.mit.edu --send-keys 7F87F9BD4D9F71A6
+```
+
+Tipical public servers required by Sonatype are: `pool.sks-keyservers.net`, `keyserver.ubuntu.com`, `keys.gnupg.net`
```
Comment (chupman) [inline RELEASING.md]: typo: Typical

File: RELEASING.md [hunk 2]
```diff
@@ -16,6 +15,38 @@ The release artifacts will be deployed into Sonatype OSS with the Maven deploy p
...
+*   4096 bits
+*   no expiration
+*   comment: `CODE SIGNING KEY`
+
+Add your key to this file:
+```Shell
+(gpg --list-sigs <key id> && gpg --armor --export <key id>) >> KEYS.
+```
+
+After that you need to publish your key to public key servers. Sometimes those key servers are down so you may want to publish to another key servers. Choose those key servers which are required by Sonatype.
```
Comment (chupman) [inline RELEASING.md]: Please move each sentence onto a separate line. I think it makes sense to group the common server list here as well.

> Once your key has been created and added to the KEYS file it needs to be published to a [public key server](https://sks-keyservers.net/status/). 
> You can upload to multiple key servers within the pool or use an alternate if `pgp.mit.edu` is down.
> Sonatype typically uses the following key servers: `pool.sks-keyservers.net`, `keyserver.ubuntu.com`, `keys.gnupg.net`

File: RELEASING.md [hunk 3]
```diff
@@ -31,6 +62,8 @@ Steps below were taken from [this guide.](https://maven.apache.org/guides/mini/g
 </settingsSecurity>
 ```
 
+**Notice:** the password usually contained in the braces `{}` and should be put with those braces into the `<master>` tag. An example is: `<master>{1234567ABCDEFG}</master>`
```
Comment (chupman) [inline RELEASING.md]: >The password and the encapsulating braces `{}` should be placed into the `<master>` tag.
>Example: `<master>{1234567ABCDEFG}</master>`

File: RELEASING.md [hunk 4]
```diff
@@ -16,6 +15,38 @@ The release artifacts will be deployed into Sonatype OSS with the Maven deploy p
...
+```Shell
+(gpg --list-sigs <key id> && gpg --armor --export <key id>) >> KEYS.
+```
+
+After that you need to publish your key to public key servers. Sometimes those key servers are down so you may want to publish to another key servers. Choose those key servers which are required by Sonatype.  
+The command to publish your keys to a key server is the next:
+```Shell
+gpg --keyserver <key server> --send-keys <key id>
+```
+An example:
```
Comment (chupman) [inline RELEASING.md]: switch to just `Example:`

File: RELEASING.md [hunk 5]
```diff
@@ -122,36 +155,61 @@ Before any artifacts can be generated and vote can be made the version number wi
...
-The current `pom.xml` files will be backed up with the name `pom.xml.releaseBackup` and left in staging.
+This command will remove `SNAPSHOT` from all versions in `pom` files.
 
 ```Shell
-mvn clean release:prepare
+mvn versions:set -DremoveSnapshot=true -DgenerateBackupPoms=false
 ```
 
-#### Create the Pull Request
+After that open the main `pom.xml` and change `<tag>HEAD</tag>` to the necessary release tag.
```
Comment (chupman) [inline RELEASING.md]: After the version has been set, update the scm tag by opening the main `pom.xml` and changing `<tag>HEAD</tag>` to match the release tag.

File: RELEASING.md [hunk 6]
```diff
@@ -122,36 +155,61 @@ Before any artifacts can be generated and vote can be made the version number wi
...
+For example: `<tag>v0.3.2</tag>`
```
Comment (chupman) [inline RELEASING.md]: Example:

File: RELEASING.md [hunk 7]
```diff
@@ -122,36 +155,61 @@ Before any artifacts can be generated and vote can be made the version number wi
...
+Create a release commit:
+```Shell
+git commit -m "JanusGraph release <version> [full build]" -s
+```
+
+After that create a release tag with the next command:
+```Shell
+git tag -a <release tag> -m ""
+```
+An example:
```
Comment (chupman) [inline RELEASING.md]: Example:

File: RELEASING.md [hunk 8]
```diff
@@ -160,32 +218,42 @@ Build Release Artifacts
...
-gpg --verify ${JG_VER}-hadoop2.zip.asc ${JG_VER}-hadoop2.zip
 ```
 
 If it fails due to Inappropriate ioctl for device error, run:
 ```Shell
 export GPG_TTY=$(tty)
 ```
 
-The `mvn clean deploy` uploads the release jars to a staging repository on Sonatype.
+*   Verify signatures validity (both commands should show good signature)
```
Comment (chupman) [inline RELEASING.md]: Verify signature validity

**Comment by porunov** [comment]
@chupman Thank you. I've applied your suggestions.

**Pull request merged**